sources/osgAnimation/BasicAnimationManager.js

```diff
@@ -135,19 +130,12 @@ define( [
             // animations to start
             this._startAnimations = {};
 
-            this._targets.length = 0;
-            this._targetsMap = {};
-
-            this._targetID.length = 0;
-
-            var i;
-            for ( i = 0; i < this._targetIDByTypes.length; i++ )
-                this._targetIDByTypes[ i ].length = 0;
+            this._resetTargets();
 
             this._activeAnimations = {};
             this._activeAnimationList.length = 0;
 
-            for ( i = 0; i < this._activeChannelsByTypes; i++ )
+            for ( var i = 0; i < this._activeChannelsByTypes; i++ )
                 this._activeChannelsByTypes[ i ].length = 0;
 
```

Comment:
In this jsperf there is not a array = []. Anyway if you do think that the best is to always do array.length = 0 ok for me. The point is to do all the same. :+1: